### PR TITLE
[AMORO-2815] Adjust docker images of apache amoro project

### LINKED ISSUE #2815: [Improvement]: Adjust docker images of apache amoro project
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

* Keep the apache/amoro, apache/amoro-flink-optimizer,
apache/amoro-spark-optimizer images.
* Drop the quick demo image.
* Drop the datanode and namenode image.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2815.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by majin1102 — APPROVED** [review]
LGTM

File: charts/amoro/values.yaml [hunk 1]
```diff
@@ -254,7 +254,7 @@ optimizer:
     properties:
       namespace: "default"
       kube-config-path: "～/.kube/config"
-      image: "arctic163/amoro:0.6"
+      image: "apache/amoro:0.6"
```
Comment (baiyangtx) [inline charts/amoro/values.yaml:257]: We shouldn't hardcore a version in the default values.yml.  I will fix this in the other PR.

File: docker/demo-cluster.sh [hunk 2]
```diff
@@ -1,186 +0,0 @@
...
-Commands:
-    start                   Setup demo cluster
-    stop                    Stop demo cluster and remove containers, volume data will be kept.
-    clean                   clean volume data.
-
-Options:
-    -v    --version         Setup Amoro image version. default is ${AMORO_TAG}
-
-EOF
-}
```
Comment (baiyangtx) [inline docker/demo-cluster.sh:46]: I think we shoud keep demo-cluster.sh but using apache/amoro and minio to create demo cluster.

Comment (baiyangtx) [reply docker/demo-cluster.sh:46]: @zhoujinsong WDYT? or we can manage demo-cluster.sh in amoro-site project.

File: docker/build.sh [hunk 3]
```diff
@@ -44,23 +42,18 @@ Usage: $0 [options] [image]
 Build for Amoro demo docker images.
 
 Images:
-    quickdemo               Build Amoro QuickStart Image, for run flink ingestion job in quick-demo http://amoro.apache.org/quick-demo/
-    namenode                Build a hadoop namenode container for quick start demo.
-    datanode                Build a hadoop datanode container for quick start demo.
     optimizer-flink         Build official Amoro optimizer deployed with flink engine for production environments.
```
Comment (zhoujinsong) [inline docker/build.sh:45]: How about changing the name to `amoro-flink-optimizer`/`amoro-spark-optimizer` here?

**Review by zhoujinsong — COMMENTED** [review]
@tcodehuber Thanks a lot for the contribution.
I left some comments, PTAL.

File: docker/demo-cluster.sh (hunk 2 above)
Comment (zhoujinsong) [reply docker/demo-cluster.sh:46]: I am not sure if we still need a shell script to create the docker-compose file or if just a compose example file is okay.

File: charts/amoro/values.yaml (hunk 1 above)
Comment (tcodehuber) [reply charts/amoro/values.yaml:257]: OK

File: docker/demo-cluster.sh (hunk 2 above)
Comment (tcodehuber) [reply docker/demo-cluster.sh:46]: I will rebuild the demo process in another PR to fix #2816, importing a docker-compose.yml

File: amoro-ams/dist/src/main/arctic-bin/conf/config.yaml [hunk 4]
```diff
@@ -142,7 +142,7 @@ containers:
 #   container-impl: com.netease.arctic.server.manager.KubernetesOptimizerContainer
 #    properties:
 #     kube-config-path: ～/.kube/config
-#     image: arctic163/amoro:0.6
+#     image: apache/amoro:0.6
```
[the remark this answers is not in the document]
Comment (tcodehuber) [reply amoro-ams/dist/src/main/arctic-bin/conf/config.yaml:145]: how about using {tag}

File: docker/build.sh (hunk 3 above)
Comment (tcodehuber) [reply docker/build.sh:45]: OK

**Review by zhoujinsong — COMMENTED** [review]
LGTM.
I prefer to wait until the new Docker Hub repository is created before merging this pull request.

**Review by czy006 — APPROVED** [review]
LGTM

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**